src/ILCompiler.Compiler/src/JitInterface/TypesDebugInfo/UserDefinedTypeDescriptor.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+using Internal.TypeSystem;
+using ILCompiler.DependencyAnalysis;
+using System.Reflection.Metadata;
+using Internal.TypeSystem.Ecma;
+
+namespace Internal.JitInterface.TypesDebugInfo.UserDefinedTypes
+{
+
+    internal struct EnumRecordTypeDescriptor
+    {
+        public ulong Value;
+        public string Name;
+    };
+
+    internal struct EnumTypeDescriptor
+    {
+        public ulong ElementType;
+        public ulong ElementCount;
+        public string Name;
+        public string UniqueName;
+    }
+
+
+    internal class UserDefinedTypeDescriptor
+    {
+        public UserDefinedTypeDescriptor(ObjectWriter objectWriter)
```

Comment:
(Maybe we would like something that we can also reuse to generate in-memory PDB files for when we use RyuJIT as an actual JIT in the future?)